src/Crypto/Lol/Cyclotomic/UCyc.hs

```diff
@@ -96,15 +96,15 @@ data UCyc t (m :: Factored) r where
 
 -- | Shorthand for frequently reused constraints that are needed for
 --  change of basis.
-type UCCtx t r = (Tensor t, CRTrans r, CRTrans (CRTExt r), CRTEmbed r,
-                  ZeroTestable r, TElt t r, TElt t (CRTExt r))
+type UCCtx t r = (Tensor t, CRTrans r, CRTrans (CRTExt r), CRTEmbed r, IntegralDomain r, IntegralDomain (CRTExt r),
```

Comment:
A type synonym that gathers all the algebraic constraints would be good here. Then it can be applied to both `r` and `CRTExt r`.